src/Workspaces/Remote/Core/ProjectSystem/WorkspaceProjectCreationInfo.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Remote.ProjectSystem
+{
+    public record WorkspaceProjectCreationInfo(string Language, string DisplayName, string? FilePath, IReadOnlyDictionary<string, string> BuildSystemProperties);
```

Comment:
>BuildSystemProperties [](http://example.com/codeflow?start=138&length=21)

Why not use EvaluationData as is? We'll need both properties and items.

Reply:
@tmat: as it is, it's not RPC friendly since the contract is you derive the type and then there's a chatty calling back and forth for each item. For properties we can easily just gather up all the properties at once and just send them over. Right now we aren't actually using this for items at all:

https://github.com/dotnet/roslyn/blob/330e1c2cee5a43d92daf477f2c54aa6d0c41a353/src/VisualStudio/Core/Def/ProjectSystem/BuildPropertyNames.cs#L39

So this seems a lot easier.

Reply:
We are going to use items. See https://github.com/dotnet/roslyn/pull/64892.

You can derive a type that has the two dictionaries in it and is serializable or create another type that has the same shape.

Reply:
> You can derive a type that has the two dictionaries in it and is serializable or create another type that has the same shape.

@tmat Yeah, the latter is the goal here. The former might _work_ but seems silly to define an RPC interface where you can only use certain derived types to actually meet the implicit contracts of the system.

Do we have any examples of items _other_ than the intermediate output path example? Because that's not a terribly motivating example given I'm not sure why it's not a property in the first place. I'd much rather just teach the project system to convert a single item group to a property when requested rather than have a bunch of infrastructure for that.

Reply:
I have other concerns with items specifically, since we have cases today where if we were going to use items more fully we'd also need to pass along some of the other metadata too, so we'd need to spec that properly.

Reply:
> I'd much rather just teach the project system to convert a single item group to a property when requested rather than have a bunch of infrastructure for that.

Sure, we can do that but what do we do when we need to read items that have multiple values? That's not such a stretch and having support for that isn't that much extra effort. It's one more immutable dict.

msbuild allows setting properties based on other properties and items based on other items during evaluation. It does not support converting items to properties, otherwise we'd just do that. If we need to read metadata of some item group we can transform them to item values. Hence, we can easily get away without supporting item metadata in the project system interface.

Reply:
> Sure, we can do that but what do we do when we need to read items that have multiple values?

What concrete examples did you have in mind?

> If we need to read metadata of some item group we can transform them to item values

Ick, the cases I had in mind where we might need this would be a nightmare for escaping. That doesn't seem like a usable approach.

If we have to ultimately add a `Dictionary<string, IEnumerable<Dictionary<string>>>` here I'm not strictly against it; it seems however the current pattern we have here isn't generally applicable for the things I'd first imagine needing to use it for. I'd rather design that hen we're actually in a place to consume that versus where we are now.